package.json

```diff
@@ -36,7 +36,8 @@
   },
   "scripts": {
     "build": "rm -rf dist && tsc -p ./tsconfig.build.json",
-    "lint": "eslint .",
+    "lint": "prettier . --check && eslint .",
+    "format": "prettier --write .",
```

Comment:
This is what I originally set out to do, but the Prettier documentation pretty [explicitly recommends](https://prettier.io/docs/en/integrating-with-linters) against this. My intention is to use `lint-staged` to automatically fix any formatting issues when committing.

Reply:
The only question is then, why do we have an eslint config if prettier runs? If we keep eslint around, then we should probably run through it, otherwise, remove it altogether? Unless prettier understands the eslint config?

Reply:
Prettier is a code formatter, and ESLint is a linter for common mistakes and code-style, but does not concern itself with formatting rules (e.g. when a line breaks, where to insert newlines/white-space). In fact the ESLint team removed several code formatting related rules in the last major to make this split of concerns more clear. They each serve a different purpose.

The way it works is that Prettier is ran first which takes care of the code formatting and then ESLint runs to take care of any other issues. I went ahead and added a `lint:fix` task that makes this more clear/convenient. Once #105 lands there will be very little reason to handle this manually, as the pre-commit hook essentially does the same.

Reply:
Got it. Thanks for explaining. I guess a lot changed in the past few years.